core/tests/test_utils.py

```diff
@@ -677,6 +678,57 @@ def save_new_valid_exploration(
         exp_services.save_new_exploration(owner_id, exploration)
         return exploration
 
+    def save_new_valid_exploration_with_states(
+            self, exploration_id, owner_id, title='A title',
+            category='A category', objective='An objective',
+            language_code=constants.DEFAULT_LANGUAGE_CODE,
+            state_names=None, interaction_ids=None):
```

Comment:
Why are state_names, interaction_ids allowed to have default values that aren't valid? Suggest requiring these to be always specified by the caller.

Reply:
Done.